CMake/typeinfo_test.cc

```diff
@@ -49,6 +49,9 @@ class type_info2 : public std::type_info
 	virtual bool __do_catch(const type_info *thrown_type,
 	                        void **thrown_object,
 	                        unsigned outer) const { return true; }
+	virtual bool __do_upcast(
```

Comment:
Do we actually need this test anymore?  I think we're now detecting the C++ runtime at run time, so we should work with any C++ runtime.

Reply:
We now work with every runtime, but we still need this test to find out whether there is a usable runtime or if we need to link the full stdlib.

Reply:
This change here is needed to prevent a dependency on symbols from the GNU runtime.